pf.go

```diff
@@ -0,0 +1,211 @@
+// +build openbsd freebsd
+
+package main
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/crowdsecurity/crowdsec/pkg/models"
+	"github.com/pkg/errors"
+
+	log "github.com/sirupsen/logrus"
+)
+
+type pfContext struct {
+	proto   string
+	table   string
+	version string
+}
+
+type pf struct {
+	inet  *pfContext
+	inet6 *pfContext
+}
+
+const (
+	backendName = "pf"
+
+	pfinetTable  = "crowdsec-blacklists"
+	pfinet6Table = "crowdsec6-blacklists"
+
+	pfctlCmd = "/sbin/pfctl"
+	pfDevice = "/dev/pf"
+
+	addBanFormat = "%s: add ban on %s for %s sec (%s)"
+	delBanFormat = "%s: del ban on %s for %s sec (%s)"
+)
+
+func newPF(config *bouncerConfig) (interface{}, error) {
+	ret := &pf{}
+
+	inetCtx := &pfContext{
+		table:   pfinetTable,
+		proto:   "inet",
+		version: "ipv4",
+	}
+
+	inet6Ctx := &pfContext{
+		table:   pfinet6Table,
+		proto:   "inet6",
+		version: "ipv6",
+	}
+
+	ret.inet = inetCtx
+
+	if config.DisableIPV6 {
+		return ret, nil
+	}
+
+	ret.inet6 = inet6Ctx
+
+	return ret, nil
+}
+
+func (ctx *pfContext) checkTable() error {
+	log.Infof("Checking pf table: %s", ctx.table)
+
+	cmd := exec.Command(pfctlCmd, "-s", "Tables")
+	out, err := cmd.CombinedOutput()
+
+	if err != nil {
+		return errors.Wrapf(err, "pfctl error : %v - %s", err, string(out))
+	} else if !strings.Contains(string(out), ctx.table) {
+		return errors.Errorf("table %s doesn't exist", ctx.table)
```

Comment:
Can we add some documentation to specify that the user needs to create both tables beforehand ?

Reply:
Yes, user should create the `pf` tables using `pf.conf`, cf.

```
# create crowdsec ipv4 table
table <crowdsec-blacklists> persist

# create crowdsec ipv6 table
table <crowdsec6-blacklists> persist
```

I will add a commit to the README.